lib/soap.js

```diff
@@ -97,6 +97,10 @@ module.exports = class Soap {
             }
             status = resultsBatch.OverallStatus;
             if (status === 'MoreDataAvailable') {
+                //as requestParams is by default optional, ensure object exists in this case
+                if (!requestParams) {
+                    requestParams = {};
+                }
```

Comment:
```suggestion
                requestParams = requestParams || {};
```
how about this instead? shorter